composer/loggers/mosaicml_logger.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+import os
+import warnings
+from queue import Queue
+from threading import Thread
+from typing import Optional
+
+import requests
+
+from composer.core.logging import LogLevel, RankZeroLoggerBackend, TLogData
+from composer.core.serializable import Serializable
+from composer.core.types import JSON, Logger, State, StateDict
+
+_MOSAICML_API_KEY_ENV = "MOSAICML_LOGGER_API_KEY"
+_MOSAICML_LOGGER_URL = "https://api.mosaicml.com/v0/log/metric"
+_JOB_ID_ENV = "MOSAICML_LOGGER_JOB_ID"
+_SWEEP_ID_ENV = "MOSAICML_LOGGER_SWEEP_ID"
+
+_STOP_LOGGING_SIGNAL = "STOP_LOGGING"
+
+log = logging.getLogger(__name__)
+
+
+def _send_data(job_id: str, sweep_id: str, data: JSON):
+    response = requests.post(_MOSAICML_LOGGER_URL,
+                             headers={"X-MosaicML-API-key": os.environ.get(_MOSAICML_API_KEY_ENV, "")},
+                             json={
+                                 "experimentID": job_id,
+                                 "runID": sweep_id,
+                                 "data": data
+                             },
+                             timeout=3)
+    return response
```

Comment:
Does the requests library automatically raise for error codes? Or do we need to add in a `response.raise_for_status()` or the like?